### PR TITLE
Update 5.1

### PR DESCRIPTION
**Pull request by Arishamays1**
## ☆Django 5.1 Update ☆
[Jira Ticket](https://anaconda.atlassian.net/browse/PKG-5242)
[Upstream](https://github.com/django/django)
# Changes
- Updated version number and `sha256`
- Skip to `Python` versions less than **3.10** due to python-support restrictions
- Added new dependencies
- Added `description` to `about` section
- Added patching for `tzdata` regarding `windows` failure

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
   name: django
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/D/Django/Django-{{ version }}.tar.gz
-  sha256: 032f8a6fc7cf05ccd1214e4a2e21dfcd6a23b9d575c6573cacc8c67828dbe642
-
+  sha256: 848a5980e8efb76eea70872fb0e4bc5e371619c70fffbe48e3e1b50b2c09455d
+  patches:
+    - 001-Win-tzdata-removal   # [win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - patches/001-Win-tzdata-removal.patch   # [win]
```

File: recipe/patches/001-Win-tzdata-removal
Comment (cbouss) [inline recipe/patches/001-Win-tzdata-removal:1]: Can you rename to 001-Win-tzdata-removal.patch?
And add a comment at the top of the file explaining why this is needed. (python-tzdata is a fallback package for systems not providing timezone data, but the conda python package depends on tzdata. Therefore we do not need to depend on python-tzdata.)

Comment (Arishamays1) [reply recipe/patches/001-Win-tzdata-removal:1]: will do! thanks!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
-  script: python -m pip install --no-deps --ignore-installed .
-  skip: True # [py<38]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed -vvv
+  skip: True  # [py<310]
   entry_points:
     - django-admin = django.core.management:execute_from_command_line
 
 requirements:
+  build:
+    - patch   # [unix]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
The patch is only applied on windows

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
+    - setuptools
     - wheel
   run:
-    - asgiref >=3.5.2,<4
-    - backports.zoneinfo # [py<39]
     - python
-    - sqlparse >=0.2.2
-    - python-tzdata # [win]
+    - asgiref >=3.8.1
+    - sqlparse >=0.3.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:32]:
```suggestion
    - sqlparse >=0.3.1
  run_constrained:
    - argon2-cffi >=19.1.0
```
See https://github.com/django/django/blob/5.1/pyproject.toml#L41C11-L41C30

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,32 +1,38 @@
...
+
+
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:35]:
```suggestion
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,32 +1,38 @@
...
 
 requirements:
+  build:
+    - patch   # [unix]
+    - m2-patch  # [win]
   host:
     - python
     - pip
-    - setuptools >=40.8.0
+    - setuptools
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - setuptools >=61.0.0,<69.3.0
```
There are upstream issues with case sensitivity in **setuptools >=69.3.0**, see:

- https://github.com/django/django/blob/5.1/pyproject.toml#L2
- https://github.com/django/django/commit/b4dd76c315ff06e4897fd380e0ad0eaf88db8a9d
- https://github.com/django/django/pull/17806#discussion_r1609743375

**Review by skupr-anaconda — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 5 above)
Comment (cbouss) [reply recipe/meta.yaml]: I wouldn't take this, it doesn't apply to our packaging and makes maintenance more difficult.

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**